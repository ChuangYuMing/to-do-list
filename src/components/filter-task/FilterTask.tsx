import React, { Dispatch, SetStateAction } from "react";
import styles from "./FilterTask.module.scss";
import { TASK_FILTER_STATUS } from "@/const.js";

type FilterTaskProps = {
  taskFilterStatus: string;
  setTaskFilterStatus: Dispatch<SetStateAction<string>>;
};

export default function AddTask({
  setTaskFilterStatus,
  taskFilterStatus,
}: FilterTaskProps) {
  const statusList = [
    TASK_FILTER_STATUS.ALL,
    TASK_FILTER_STATUS.UNCOMPLETED,
    TASK_FILTER_STATUS.COMPLETED,
  ].map((status) => (
    <span
      key={status}
      data-status={status}
      className={taskFilterStatus === status ? styles["is-active"] : ""}
    >
      {status}
    </span>
  ));

  const handleFilter = (e: React.MouseEvent<HTMLDivElement>) => {
    const target = e.target as HTMLDivElement;
    const status = target.dataset.status;

    if (status) {
      setTaskFilterStatus(status);
    }
  };

  return (
    <div className={styles["filter-task-wrap"]} onClick={handleFilter}>
      {statusList}
    </div>
  );
}
