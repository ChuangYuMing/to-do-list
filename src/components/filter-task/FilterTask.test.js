import React from "react";
import { render } from "@testing-library/react";
import FilterTask from "./FilterTask";

it("renders correctly", () => {
  const { asFragment } = render(<FilterTask />);
  expect(asFragment()).toMatchSnapshot();
});
