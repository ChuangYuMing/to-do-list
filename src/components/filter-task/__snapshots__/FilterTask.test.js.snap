// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="filter-task-wrap"
  >
    <span
      class=""
      data-status="All"
    >
      All
    </span>
    <span
      class=""
      data-status="Uncompleted"
    >
      Uncompleted
    </span>
    <span
      class=""
      data-status="Completed"
    >
      Completed
    </span>
  </div>
</DocumentFragment>
`;
